python/pyspark/sql/session.py

```diff
@@ -952,12 +953,30 @@ def createDataFrame(  # type: ignore[misc]
             schema = [x.encode("utf-8") if not isinstance(x, str) else x for x in schema]
 
         try:
-            import pandas
+            import pandas as pd
 
             has_pandas = True
         except Exception:
             has_pandas = False
-        if has_pandas and isinstance(data, pandas.DataFrame):
+
+        try:
+            import numpy as np
```

Comment:
All NumPy versions should be fine as long as pandas can convert the ndarray to pandas Frame.